docs/sdk/serverless-signing/overview.md

```diff
@@ -2,42 +2,73 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+## Example Lit Action Implementation
 
-This logic is encoded using a Lit Action, an immutable JavaScript program that can be "assigned" to the key pairs generated on Lit and used to dictate how they are used.
+To illustrate the power and flexibility of Lit Actions, let's consider a practical example:
 
-A simple example would be a Lit Action that checks if a number is prime. To start, you would generate a [Programmable Key Pair](../../user-wallets/pkps/overview.md) and assign it to your Lit Action so that it could use it to produce a signature. This signature would only be returned if the conditions defined within your Lit Action were met. In this case, the Lit Action would verify that the input was indeed a prime number. Every Lit node would execute your program independently and provision their key share down to the requesting client. Only after more than two-thirds of these shares have been collected can the complete signature be formed. You can read more about how Lit works [here](../../resources/how-it-works.md).
+ > A Lit Action that signs a transaction only if the reported temperature for a specific area is below a defined threshold.
 
-## Features and Examples
+###### How it would work:
 
-### Features
+1. The Lit Action fetches temperature data from three different weather APIs.
+   - The choice of using three APIs is arbitrary, but demonstrates how data from multiple sources can be fetched from within a single Lit Action.
+2. It calculates the average temperature from the these sources.
+3. If the average temperature is below a predefined threshold, the Lit Action uses a Programmable Key Pair (PKP) to sign a transaction that transfers tokens on a blockchain.
+4. The signed transaction can be broadcasted to the blockchain network immediately, or returned for later submission.
 
-- **[Blockchain Agnostic](../../resources/supported-chains.md)**: Lit Actions can be used to read and write data to or even between blockchains with Programmable Key Pairs
-- **Immutable**: Once a Lit Action has been published, it cannot be modified, just like a smart contract deployed on a blockchain. The [Mint/Grant/Burn](https://github.com/LIT-Protocol/js-sdk/blob/70a041a97b56ba1a75724ba2cd56952b622e8a7f/packages/contracts-sdk/src/abis/PKPNFT.ts#L376) function allows you to atomically link a PKP to an authorized set of Lit Actions, guaranteeing that a particular PKP can only ever be used to sign data from within the approved set.
-- **Off-Chain Compatibility**: You can make arbitrary HTTP requests from a Lit Action, meaning you can pull in data from off-chain sources natively, without needing to use of a third party oracle.
+###### This example showcases how Lit Actions can:
 
-### Examples
+- Interact with on and off-chain APIs/RPCs
+- Perform computations and make decisions using fetched data
+- Use PKPs for conditional signing
```

Comment:
we could also link to the conditional signing section here where it is referenced